support/conditions/conditions.go

```diff
@@ -39,8 +39,10 @@ func ExpectedHCConditions(hostedCluster *hyperv1.HostedCluster) map[hyperv1.Cond
 
 		conditions[hyperv1.ValidAWSIdentityProvider] = metav1.ConditionTrue
 		conditions[hyperv1.AWSDefaultSecurityGroupCreated] = metav1.ConditionTrue
-		conditions[hyperv1.AWSEndpointAvailable] = metav1.ConditionTrue
-		conditions[hyperv1.AWSEndpointServiceAvailable] = metav1.ConditionTrue
+		if hostedCluster.Spec.Platform.AWS.EndpointAccess == hyperv1.Private {
```

Comment:
publicAndPrivate topology must also satisfy this.

Reply:
addressed